test/unit-test/ota_cbor_utest.c

```diff
@@ -101,34 +110,268 @@ void test_OTA_CborDecodeStreamResponse()
+        NULL,                              /* block bitmap. */
+        sizeof( bitmap ),                  /* size of bitmap. */
+        otaconfigMAX_NUM_BLOCKS_REQUEST ); /* number of block requested. */
+
+    TEST_ASSERT_FALSE( result );
+
+    /* Test that encoding fails with invalid message buffer. */
+    result = OTA_CBOR_Encode_GetStreamRequestMessage(
+        NULL,                              /* output message buffer. */
+        sizeof( cborWork ),                /* size of output message buffer. */
+        &encodedSize,                      /* size of encoded message. */
+        CBOR_TEST_CLIENTTOKEN_VALUE,       /* client token. */
+        1,                                 /* file id. */
+        OTA_FILE_BLOCK_SIZE,               /* block size. */
+        0,                                 /* block offset. */
+        ( uint8_t * ) &bitmap,             /* block bitmap. */
+        sizeof( bitmap ),                  /* size of bitmap. */
+        otaconfigMAX_NUM_BLOCKS_REQUEST ); /* number of block requested. */
+    TEST_ASSERT_FALSE( result );
+
+    /* Test that encoding fails with invalid size of encoded message. */
+    result = OTA_CBOR_Encode_GetStreamRequestMessage(
+        cborWork,                          /* output message buffer. */
+        sizeof( cborWork ),                /* size of output message buffer. */
+        NULL,                              /* size of encoded message. */
+        CBOR_TEST_CLIENTTOKEN_VALUE,       /* client token. */
+        1,                                 /* file id. */
+        OTA_FILE_BLOCK_SIZE,               /* block size. */
+        0,                                 /* block offset. */
+        ( uint8_t * ) &bitmap,             /* block bitmap. */
+        sizeof( bitmap ),                  /* size of bitmap. */
+        otaconfigMAX_NUM_BLOCKS_REQUEST ); /* number of block requested. */
+    TEST_ASSERT_FALSE( result );
+
+    /* Test that encoding fails with invalid client token. */
+    result = OTA_CBOR_Encode_GetStreamRequestMessage(
+        cborWork,                          /* output message buffer. */
+        sizeof( cborWork ),                /* size of output message buffer. */
+        &encodedSize,                      /* size of encoded message. */
+        NULL,                              /* client token. */
+        1,                                 /* file id. */
+        OTA_FILE_BLOCK_SIZE,               /* block size. */
+        0,                                 /* block offset. */
+        ( uint8_t * ) &bitmap,             /* block bitmap. */
+        sizeof( bitmap ),                  /* size of bitmap. */
+        otaconfigMAX_NUM_BLOCKS_REQUEST ); /* number of block requested. */
+    TEST_ASSERT_FALSE( result );
+}
+
+/**
+ * @brief Test OTA_CBOR_Decode fails for invalid(NULL) parameters or
+ * incorrect datatype of a field.
+ *
+ */
+void test_OTA_CborDecodeStreamResponse_Invalid()
```

Comment:
It looks like this unit test isn't covering the case where pMessageBuffer is NULL. Would this fall under this unit test?

Reply:
It looks like this check was missing in the library as well, thanks for pointing this out. Added check in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/150/commits/60921df55008d5a2df8709f8975aafb7d6c42040